pidg.go

```diff
@@ -0,0 +1,108 @@
+package idg
+
+import (
+	"io"
+	"os"
+	"path"
+
+	"github.com/PathDNA/atoms"
+	"github.com/itsmontoya/mum"
+)
+
+//NewPersistent will return a new ID generator
+func NewPersistent(key, dir string) (pidg *PIDG, err error) {
```

Comment:
Not really a huge fan of having 1 file per key but I can't think of anything better right now.

Reply:
I feel like from a speed and simplicity standpoint, there is nothing better. Additionally, our services will utilize 20 indexes at max (being very generous with this estimate). We could easily make a data/indexes directory.